### PR TITLE
qemu.tests: Update pci_hotplug to cover more scenario

### PR DESCRIPTION
**Pull request by spcui**
Signed-off-by: Shuping Cui <scui@redhat.com>

ID: 1081684 1081682

### CODE REVIEW

**Comment by CongSmile** [comment]
Hi @spcui 

Why not the cfg to block_hotplug.cfg which is more clear?

Thanks.

**Comment by vipmike007** [comment]
Review Failed 
virt.qemu.smp_2.4096m.repeat1.run_test.Host_RHEL.7.2.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.pci_hotplug.hotplug_nic.nic_rtl8139.with_system_reset                 FAIL       30       Unhandled IndexError: list index out of range    [context: start hot-deleting pci device, repeat 0]
virt.qemu.smp_2.4096m.repeat1.run_test.Host_RHEL.7.2.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.pci_hotplug.hotplug_nic.nic_rtl8139.with_block_resize                 ERROR      33       Can not find the block size for the deivce. The output of command is:     [context: Running sub test 'block_resize' before hotunplug --> Check image size in guest]
virt.qemu.smp_2.4096m.repeat1.run_test.Host_RHEL.7.2.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.pci_hotplug.hotplug_nic.nic_virtio_net.with_reboot                    FAIL       30       Unhandled IndexError: list index out of range    [context: start hot-deleting pci device, repeat 0]
virt.qemu.smp_2.4096m.repeat1.run_test.Host_RHEL.7.2.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.pci_hotplug.hotplug_nic.nic_virtio_net.with_shutdown                  FAIL       30       Unhandled IndexError: list index out of range    [context: start hot-deleting pci device, repeat 0]
virt.qemu.smp_2.4096m.repeat1.run_test.Host_RHEL.7.2.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.pci_hotplug.hotplug_nic.nic_virtio_net.with_stop_continue             FAIL       30       Unhandled IndexError: list index out of range    [context: start hot-deleting pci device, repeat 0]
virt.qemu.smp_2.4096m.repeat1.run_test.Host_RHEL.7.2.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.pci_hotplug.hotplug_nic.nic_virtio_net.with_system_reset              FAIL       30       Unhandled IndexError: list index out of range    [context: start hot-deleting pci device, repeat 0]
virt.qemu.smp_2.4096m.repeat1.run_test.Host_RHEL.7.2.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.pci_hotplug.hotplug_nic.nic_virtio_net.with_block_resize              ERROR      32       Can not find the block size for the deivce. The output of command is:     [context: Running sub test 'block_resize' before hotunplug --> Check image size in guest]
virt.qemu.smp_2.4096m.repeat1.run_test.Host_RHEL.7.2.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.pci_hotplug.hotplug_nic.nic_e1000.with_reboot                         FAIL       30       Unhandled IndexError: list index out of range    [context: start hot-deleting pci device, repeat 0]
virt.qemu.smp_2.4096m.repeat1.run_test.Host_RHEL.7.2.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.pci_hotplug.hotplug_nic.nic_e1000.with_shutdown                       FAIL       29       Unhandled IndexError: list index out of range    [context: start hot-deleting pci device, repeat 0]
virt.qemu.smp_2.4096m.repeat1.run_test.Host_RHEL.7.2.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.pci_hotplug.hotplug_nic.nic_e1000.with_stop_continue                  FAIL       30       Unhandled IndexError: list index out of range    [context: start hot-deleting pci device, repeat 0]
virt.qemu.smp_2.4096m.repeat1.run_test.Host_RHEL.7.2.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.pci_hotplug.hotplug_nic.nic_e1000.with_system_reset                   FAIL       30       Unhandled IndexError: list index out of range    [context: start hot-deleting pci device, repeat 0]
virt.qemu.smp_2.4096m.repeat1.run_test.Host_RHEL.7.2.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.pci_hotplug.hotplug_nic.nic_e1000.with_block_resize                   ERROR      84       Can not find the block size for the deivce. The output of command is:     [context: Running sub test 'block_resize' before hotunplug --> Check image size in guest]

**Comment by spcui** [comment]
Hi @CongSmile @vipmike007 
updated with your comments.

**Comment by CongSmile** [comment]
Hi @spcui 

Could you please tested the code with avocado-vt?

There are some errors when tested with avocado-vt.

Thanks.

File: generic/tests/cfg/block_hotplug.cfg [hunk 1]
```diff
@@ -16,6 +16,13 @@
...
+        # boot without data disk -> hotplug a disk -> do something -> hotunplug disk
+        - @default:
+            need_hotplug = yes
+        # boot with a data disk -> do something -> hotunplug disk
+        - unplug:
+            need_hotplug = no
+    variants:
         - fmt_qcow2:
             image_extra_params = ""
             image_format_stg = qcow2
```
Comment (suqinhuang) [inline generic/tests/cfg/block_hotplug.cfg]: Can we remove this variants, it's a general option

**Comment by suqinhuang** [comment]
need to re-write this patch, close it

**Pull request closed without merging**